src/components/spreadsheet/custom-spreadsheet/spreadsheet-collection-save-dialog.tsx

```diff
@@ -0,0 +1,248 @@
+import {
+    SpreadsheetConfig,
+    SpreadsheetCollection,
+    ColumnDefinitionDto,
+    SpreadsheetEquipmentType,
+} from '../config/spreadsheet.type';
+import { v4 as uuid4 } from 'uuid';
+import { saveSpreadsheetCollection } from '../../../services/explore';
+
+interface SpreadsheetCollectionSaveDialogProps {
+    open: UseStateBooleanReturn;
+}
+
+const styles = {
+    checkboxSelectAll: (theme: Theme) => ({
+        padding: theme.spacing(0, 3, 2, 2),
+        fontWeight: 'bold',
+        cursor: 'pointer',
+    }),
+    checkboxItem: {
+        cursor: 'pointer',
+        padding: '0 16px',
+    },
+};
+
+interface TableState {
+    name: string;
+    type: SpreadsheetEquipmentType;
+    selected: boolean;
+    index: number;
+}
+
+export const SpreadsheetCollectionSaveDialog: FunctionComponent<SpreadsheetCollectionSaveDialogProps> = ({ open }) => {
+    const { snackError, snackInfo } = useSnackMessage();
+    const intl = useIntl();
+    const tables = useSelector((state: AppState) => state.tables.definitions);
+    const columnsStates = useSelector((state: AppState) => state.tables.columnsStates);
+
+    const [localTablesState, setLocalTablesState] = useState<TableState[]>([]);
+    const [selectedConfigs, setSelectedConfigs] = useState<SpreadsheetConfig[]>([]);
+    const [showElementCreationDialog, setShowElementCreationDialog] = useState(false);
+
+    // Initialize local state when dialog opens
+    useEffect(() => {
+        if (open.value) {
+            setLocalTablesState(
+                tables.map((table, index) => ({
+                    name: table.name,
+                    type: table.type,
+                    selected: true,
+                    index,
+                }))
+            );
+        }
+    }, [open.value, tables]);
+
+    const handleClose = useCallback(() => {
+        open.setFalse();
+    }, [open]);
+
+    const handleToggle = useCallback(
+        (index: number) => () => {
+            setLocalTablesState((prev) =>
+                prev.map((table) => (table.index === index ? { ...table, selected: !table.selected } : table))
+            );
+        },
+        []
+    );
+
+    const handleToggleAll = useCallback(() => {
+        setLocalTablesState((prev) => {
+            const allSelected = prev.every((table) => table.selected);
+            return prev.map((table) => ({
+                ...table,
+                selected: !allSelected,
+            }));
+        });
+    }, []);
+
+    const handleDrag = useCallback(({ source, destination }: DropResult) => {
+        if (destination) {
+            setLocalTablesState((prev) => {
+                const reordered = [...prev];
+                const [removed] = reordered.splice(source.index, 1);
+                reordered.splice(destination.index, 0, removed);
+                return reordered;
+            });
+        }
+    }, []);
+
+    const getReorderedColumns = useCallback(
```

Comment:
```suggestion
    const getTableColumns = useCallback(
```
Why call it "reordered" ? We return the natural columns order ...